### ISSUE TITLE
FP16, Cuda Error in findFastestTactic: 700 (an illegal memory access was encountered)

### ISSUE THREAD

**Issue by DonggeunYu** [NVIDIA/TensorRT#1098]
## Description
@ttyio 
Model is about object detection.
The FP32 converted and worked fine.
If use FP16, But got error.

## Environment

**TensorRT Version**: 7.2.1
**NVIDIA GPU**: V100
**NVIDIA Driver Version**: 10.2
**CUDA Version**: 10.2
**CUDNN Version**: 10.2
**Operating System**: 
**Python Version (if applicable)**: 3.8
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.7.1
**Baremetal or Container (if so, version)**: 

'''
[TensorRT] VERBOSE: *************** Autotuning format combination: Half(1,5), Int32(1) -> Half(1,5) ***************
[TensorRT] VERBOSE: --------------- Timing Runner: Gather_2189 (Gather)
[TensorRT] VERBOSE: Tactic: 1 time 0.037888
[TensorRT] VERBOSE: Tactic: 2 time 0.067584
[TensorRT] VERBOSE: Tactic: 3 time 0.114688
[TensorRT] VERBOSE: Tactic: 4 time 0.013312
[TensorRT] VERBOSE: Tactic: 6 time 0.014336
[TensorRT] VERBOSE: Fastest Tactic: 4 Time: 0.013312
[TensorRT] VERBOSE: *************** Autotuning format combination: Float(1,256,65536,16777216), Float(1,5) -> Float(1,7,49,12544) ***************
[TensorRT] VERBOSE: --------------- Timing Runner: (Unnamed Layer* 2832) [PluginV2DynamicExt] (PluginV2)
[TensorRT] VERBOSE: Tactic: 0 time 0.662528
[TensorRT] VERBOSE: Fastest Tactic: 0 Time: 0.662528
[TensorRT] VERBOSE: *************** Autotuning format combination: Half(1,256,65536,16777216), Half(1,5) -> Half(1,7,49,12544) ***************
[TensorRT] VERBOSE: --------------- Timing Runner: (Unnamed Layer* 2832) [PluginV2DynamicExt] (PluginV2)
[TensorRT] ERROR: ../builder/cudnnBuilderUtils.cpp (426) - Cuda Error in findFastestTactic: 700 (an illegal memory access was encountered)
[TensorRT] ERROR: ../rtSafe/safeRuntime.cpp (32) - Cuda Error in free: 700 (an illegal memory access was encountered)
terminate called after throwing an instance of 'nvinfer1::CudaError'
  what():  std::exception
'''

**Comment by ttyio** [comment]
Hello @DonggeunYu , I see the failure come from the plugin. Did you use any custom plugin? thanks!

**Comment by DonggeunYu** [comment]
Right. I used custom plugin.

**Comment by ttyio** [comment]
@DonggeunYu , seems there is some issue in the plugin FP16 implementation.
Try cuda-memcheck it can tell more detail about the kernel caused illegal memory access.

**Comment by DonggeunYu** [comment]
WOW, Thank you @ttyio.

**Comment by DonggeunYu** [comment]
@ttyio
I found problem custom op. but I don't know how to fix it anymore.
If i send problem custom op code by your email, can you see my code?

**Comment by ttyio** [comment]
Hello @DonggeunYu , I have replied in your email, thanks

**Comment by DonggeunYu** [comment]
Oh, Thanks. There was such a problem...

**Comment by DonggeunYu** [comment]
It wasn't an custom operator's problem.

TensorRT Output:
[TensorRT] VERBOSE: --------------- Timing Runner: (Unnamed Layer* 2836) [PluginV2DynamicExt] (PluginV2)
[TensorRT] ERROR: ../builder/cudnnBuilderUtils.cpp (426) - Cuda Error in findFastestTactic: 700 (an illegal memory access was encountered)
[TensorRT] ERROR: ../rtSafe/safeRuntime.cpp (32) - Cuda Error in free: 700 (an illegal memory access was encountered)
terminate called after throwing an instance of 'nvinfer1::CudaError'

ONNX VERVOSE:
%2836 : Long(requires_grad=0, device=cpu) = onnx::Constant[value={0}]()
  %2837 : Float(2000:5, requires_grad=1, device=cuda:0) = onnx::Gather[axis=1](%2812, %2836)

PyToch code:
dx = deltas[:, 0] # deltas is [2000, 5] size

An error occurs when creating a ONNX:Constant and receiving it as a Long.

**Comment by ttyio** [comment]
Hello @DonggeunYu , could you use netron to check the datatype `long` is represent as `INT32` in ONNX, onnx parser is OK to handle the `INT32` datatype.  
Or did I miss something, could you elaborate more on why it was not custom operator's problem? thanks

**Comment by DonggeunYu** [comment]
@ttyio Yes, The custom plugin problem was correct and fixed.
Thank you ~!

**Comment by p890040** [comment]
> @DonggeunYu , seems there is some issue in the plugin FP16 implementation.
> Try cuda-memcheck it can tell more detail about the kernel caused illegal memory access.

Hi. I also encounter this issue for a customized plugin, but I have no idea how to fix it.
Is there any example or document about this issue?

**Comment by DonggeunYu** [comment]
@p890040  Actually, it was a problem that the FP16 custom layer was wrongly written.
